tests/processor/test_tasks.py

```diff
@@ -146,18 +160,84 @@ def test_get_report_files_second_task_not_called(self,
+    @patch('masu.processor.tasks.ReportStatsDBAccessor.get_last_completed_datetime')
+    @patch('masu.processor.tasks.ReportStatsDBAccessor.get_last_started_datetime')
+    @patch('masu.processor.tasks._get_report_files')
+    @patch('masu.processor.tasks.process_report_file')
+    def test_get_report_files_timestamps_aligned(self,
+                                                 mock_process_files,
+                                                 mock_get_files,
+                                                 mock_started,
+                                                 mock_completed):
+        """
+        Test that the chained task is called when start timestamp is before
+        end timestamp.
+        """
+        mock_process_files.delay = Mock()
+        mock_get_files.return_value = self.fake_reports
+
+        mock_started.return_value = self.yesterday
+        mock_completed.return_value = self.today
+
+        get_report_files(**self.fake_get_report_args)
+        mock_process_files.delay.assert_called()
+
+    @patch('masu.processor.tasks.ReportStatsDBAccessor.get_last_completed_datetime')
+    @patch('masu.processor.tasks.ReportStatsDBAccessor.get_last_started_datetime')
+    @patch('masu.processor.tasks._get_report_files')
+    @patch('masu.processor.tasks.process_report_file')
+    def test_get_report_files_timestamps_misaligned(self,
+                                                    mock_process_files,
+                                                    mock_get_files,
+                                                    mock_started,
+                                                    mock_completed):
+        """
+        Test that the chained task is called when start timestamp is before
+        end timestamp.
+        """
+        mock_process_files.delay = Mock()
+        mock_get_files.return_value = self.fake_reports
+
+        mock_started.return_value = self.today
+        mock_completed.return_value = self.yesterday
+
+        get_report_files(**self.fake_get_report_args)
         mock_process_files.delay.assert_not_called()
 
+    @patch('masu.processor.tasks.ReportStatsDBAccessor.get_last_completed_datetime')
+    @patch('masu.processor.tasks.ReportStatsDBAccessor.get_last_started_datetime')
+    @patch('masu.processor.tasks._get_report_files')
+    @patch('masu.processor.tasks.process_report_file')
+    def test_get_report_files_timestamps_empty(self,
+                                               mock_process_files,
+                                               mock_get_files,
+                                               mock_started,
+                                               mock_completed):
+        """
+        Test that the chained task is called when start timestamp is before
+        end timestamp.
+        """
+        mock_process_files.delay = Mock()
+        mock_get_files.return_value = self.fake_reports
+
+        mock_started.return_value = None
```

Comment:
Do we need to be concerned with the cases where something is completed, but was never started, or was started but never completed? I know we said that the latter can be addressed by the next time the task is run.

Reply:
AFAICS, there's no possibility for a "completed but never started" case.

There is the possibility of "started, but error causes no completion". Right now, that is not handled at all. It will require admin intervention - the simplest solution is updating the `last_completed_datetime` filed to `now()`.

Reply:
Only asked because the test had handling for those cases.

Reply:
Even though it's a crazy situation, things do happen and it's always nice to know how new logic will react under these unexpected situations.  Because the main `if` condition in this PR is pretty complicated, having a test to ensure each part through it is covered is a worthy goal.